python/pyarrow/tests/strategies.py

```diff
@@ -323,9 +323,15 @@ def arrays(draw, type, size=None, nullable=True):
         value = st.datetimes(timezones=st.just(tz), min_value=min_datetime,
                              max_value=max_datetime)
     elif pa.types.is_duration(ty):
-        value = st.timedeltas()
+        value = st.timedeltas(
+            min_value=datetime.timedelta(days=-96075),
+            max_value=datetime.timedelta(days=96075)
+        )
     elif pa.types.is_interval(ty):
-        value = st.timedeltas()
+        value = st.timedeltas(
+            min_value=datetime.timedelta(days=-96075),
```

Comment:
So roughly 106750 (~292 years) should be the max but it fails with that max. I reduced 10% for now.